exonum-java-proofs/src/main/java/com/exonum/binding/storage/proofs/map/flat/ProofStatus.java

```diff
@@ -24,7 +24,8 @@
   NON_TERMINAL_NODE("Proof entry in a singleton proof is of branch type (must be a leaf)"),
   INVALID_STRUCTURE("Proof has invalid structure"),
   INVALID_ORDER("Proof entries are placed in the wrong order"),
-  DUPLICATE_PATH("There are entries with duplicate keys");
+  DUPLICATE_PATH("There are entries with duplicate keys"),
+  EMBEDDED_PATH("Some keys in a proof are prefixes of others");
```

Comment:
Shall we separate these two errors:
* A key in the proof is a prefix of the key of another proof entry.
* A key in the proof is a prefix of a requested key.

Combining them makes a difficult-to-understand description.

BTW, `INVALID_STRUCTURE` is no longer used.

Reply:
I don't think we should - these are both cases of the same error, they don't differ in any significant way. In rust it's the same `EmbeddedPaths` error.
I could change the description to the one we use in rust - `One path in the proof is a prefix of another path.` - would that be better?

Reply:
They differ in the source of a key that has a prefix in `proofs` (and the call stack) — but we can use a more generic description.